plugins/automation/app/models/discourse_automation/field.rb

```diff
@@ -12,12 +12,16 @@ class Field < ActiveRecord::Base
     around_save :on_update_callback
 
     def on_update_callback
+      automation.fields.reload
+
       previous_fields = automation.serialized_fields
 
       automation.reset!
 
       yield
 
+      automation.fields.reload
+
```

Comment:
This is not idea, but made complicated by the current upsert_field strategy. Possible improvements involve:

- having the callback on automation and not field
- calling automation.save when calling upsert_field inspecs